src/ts/background.js

```diff
@@ -1,19 +1,16 @@
 import jq from 'jq-web/jq.wasm.js';
 
-function getJq(jsonInput, filter = '.') {
-    return jq(JSON.parse(jsonInput), filter);
-}
-
 // Build the connection to the frontend
 chrome.runtime.onConnect.addListener(port => {
     // Check the connection
     if (port.name === 'jq-backend-connection') {
         // Listen to messages from frontend
         port.onMessage.addListener(msg => {
             // Filter the JSON with JQ
-            const jqResult = JSON.stringify(getJq(msg.json, msg.filter));
-            // Send the result from JQ to the frontend
-            port.postMessage({jqResult: jqResult});
+            jq.promised.raw(msg.json, msg.filter).then(result => {
+              // Send the result from JQ to the frontend
+              port.postMessage({jqResult: jqResult})
+            }).catch(err => console.error(err));
```

Comment:
I believe console.error in a background thread does nothing. 
It would better to `port.postMessage({error: err})` so that it can be logged or shown in a context available to the user.

This isn't a regression from the existing code though so I don't think you need to make that change I just wanted to note it.